src/market/market.module.ts

```diff
@@ -431,13 +431,17 @@ export class MarketModuleImpl implements MarketModule {
     return this.publishAndRefreshDemand(options.demandSpecification).pipe(
       // For each fresh demand, start to watch the related market conversation events
       switchMap((freshDemand) => this.collectMarketProposalEvents(freshDemand)),
-      // Pluck the proposal received events, notify about the rest
-      switchMap((event) => this.mapToOfferProposalAndEmitEvents(event)),
-      // Issue debug message
-      tap((proposal) => this.logger.debug("Received offer proposal from market", { proposal })),
-      // We are interested only in Initial and Draft proposals
-      filter((proposal) => proposal.isInitial() || proposal.isDraft()),
-      // batch initial proposals and  deduplicate them by provider key, pass-though proposals in other states
+      // Notify of proposals event
+      tap((event) => this.emitMarketProposalEvents(event)),
```

Comment:
I think this should be moved to inside `collectMarketProposalEvents`, now if we (or the user) subscribe to proposal events using `collectMarketProposalEvents` then they won't be emitted in the event emitter.

```ts
  collectMarketProposalEvents(demand: Demand): Observable<MarketProposalEvent> {
    return this.deps.marketApi.collectMarketProposalEvents(demand).pipe(
      tap((event) => this.logger.debug("Received demand offer event from yagna", { event })),
      tap((event) => this.emitMarketProposalEvents(event)),
    );
  }
```